resources/testharness.js

```diff
@@ -1706,6 +1706,62 @@
     }
     expose_assert(assert_array_approx_equals, "assert_array_approx_equals");
 
+    /**
+     * Get bitwise of the given value.
+     * @param {number} value
+     * @param {string} dataType A data type string, like "float32", "int8",
+     *     more data type strings, please see:
+     *     https://webmachinelearning.github.io/webnn/#enumdef-mloperandtype
+     * @return {number} A 64-bit signed integer.
+     */
+    function getBitwise(value, dataType) {
+        const buffer = new ArrayBuffer(8);
+        const int64Array = new BigInt64Array(buffer);
+        int64Array[0] = value < 0 ? ~BigInt(0) : BigInt(0);
+        let typedArray;
+
+        if (dataType === "float32") {
+            typedArray = new Float32Array(buffer);
+        } else {
+            throw new AssertionError(`Data type ${dataType} is not supported`);
+        }
+
+        typedArray[0] = value;
+
+        return int64Array[0];
+    }
+
+    /**
+     * Assert that each array property in ``actual`` is a number being close enough to the corresponding
+     * property in ``expected`` by the acceptable ULP distance ``nulp`` with given ``dataType`` data type.
+     *
+     * @param {string} op
+     * @param {Array} actual - Array of test values.
+     * @param {Array} expected - Array of values expected to be close to the values in ``actual``.
+     * @param {number} [nulp=0] - A BigInt value indicates acceptable ULP distance, default 0.
+     * @param {string} [dataType="float32"] - A data type string, default "float32",
+     *     more data type strings, please see:
+     *     https://webmachinelearning.github.io/webnn/#enumdef-mloperandtype
+     */
+    function assert_array_approx_equals_ulp(actual, expected, nulp, dataType)
```

Comment:
Hi @sideshowbarker, I want to introduce this new `assert_array_approx_equals_ulp` assert into `resources/testharness.js` for testing operations tests of [WebNN API](https://webmachinelearning.github.io/webnn/) , would you please help review this change, thanks.

Reply:
Hi @gsnedders, would you please take a look at this update of  `resources/testharness.js` with new `assert_array_approx_equals_ulp` assert, thanks.

Reply:
Hi @dontcallmedom, It's been over one month, I'm still not receive any response from WPT infra team about this new assertion, could you help make a bridge? Thanks.